pkg/controller/sparkapplication/controller.go

```diff
@@ -258,91 +259,88 @@ func (c *Controller) processNextItem() bool {
 	return true
 }
 
-// Helper data structure to encapsulate current state of the driver pod.
-type driverState struct {
-	podName            string         // Name of the driver pod.
-	sparkApplicationID string         // Spark application ID.
-	nodeName           string         // Name of the node the driver pod runs on.
-	podPhase           apiv1.PodPhase // Driver pod phase.
-	completionTime     metav1.Time    // Time the driver completes.
-	err                error          // Error if any found.
-
-}
-
 func (c *Controller) getAppPods(app *v1beta1.SparkApplication) ([]*apiv1.Pod, error) {
 	// Fetch all the pods for the current run of the application.
 	selector := labels.SelectorFromSet(labels.Set(getResourceLabels(app)))
 	pods, err := c.podLister.Pods(app.Namespace).List(selector)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get pods for SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
 	}
+
 	return pods, nil
 }
 
+// Fetch driver pod from APIServer.
+func (c *Controller) getDriverPodFromApiServer(app *v1beta1.SparkApplication) (*apiv1.Pod, error) {
+	selector := labels.SelectorFromSet(map[string]string{config.SparkAppNameLabel: app.Name, config.SparkRoleLabel: config.SparkDriverRole})
+	listOptions := metav1.ListOptions{LabelSelector: selector.String()}
+	podsList, err := c.kubeClient.CoreV1().Pods(app.Namespace).List(listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get driver pod for SparkApplication from APIServer %s/%s: %v", app.Namespace, app.Name, err)
```

Comment:
Suggesting changing the error message to `"failed to get driver pod for SparkApplication %s%s from the API server"`.